lib/class-wp-rest-pattern-directory-controller.php

```diff
@@ -0,0 +1,262 @@
+			'categories'     => array_map( 'sanitize_title', $raw_pattern->category_slugs ),
+			'keywords'       => array_map( 'sanitize_title', $raw_pattern->keyword_slugs ),
+			'description'    => sanitize_text_field( $raw_pattern->meta->wpop_description ),
+			'viewport_width' => absint( $raw_pattern->meta->wpop_viewport_width ),
+		);
+
+		return $prepared_pattern;
+	}
+
+	/**
+	 * Retrieves the pattern's schema, conforming to JSON Schema.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$this->schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'pattern-directory-item',
+			'type'       => 'object',
+
+			'properties' => array(
+				'id'             => array(
+					'description' => __( 'The pattern ID.', 'gutenberg' ),
+					'required'    => true,
+					'type'        => 'integer',
+					'minimum'     => 1,
+					'context'     => array( 'view' ),
+				),
+
+				'title'          => array(
+					'description' => __( 'The pattern title, in human readable format.', 'gutenberg' ),
+					'required'    => true,
+					'type'        => 'string',
+					'minLength'   => 1,
+					'context'     => array( 'view' ),
+				),
+
+				'content'        => array(
+					'description' => __( 'The pattern content.', 'gutenberg' ),
+					'required'    => true,
+					'type'        => 'string',
+					'minLength'   => 1,
+					'context'     => array( 'view' ),
+				),
+
+				'categories'     => array(
+					'description' => __( "The pattern's category slugs.", 'gutenberg' ),
+					'required'    => true,
+					'type'        => 'array',
+					'uniqueItems' => true,
+					'items'       => array( 'type' => 'string' ),
+					'context'     => array( 'view' ),
+				),
+
+				'keywords'       => array(
+					'description' => __( "The pattern's keyword slugs.", 'gutenberg' ),
+					'required'    => true,
+					'type'        => 'array',
+					'uniqueItems' => true,
+					'items'       => array( 'type' => 'string' ),
+					'context'     => array( 'view' ),
+				),
+
+				'description'    => array(
+					'description' => __( 'A description of the pattern.', 'gutenberg' ),
+					'required'    => true,
+					'type'        => 'string',
+					'minLength'   => 1,
+					'context'     => array( 'view' ),
+				),
+
+				'viewport_width' => array(
+					'description' => __( 'The preferred width of the viewport when previewing a pattern, in pixels.', 'gutenberg' ),
+					'type'        => 'integer',
+					'required'    => true,
+					'context'     => array( 'view' ),
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Retrieves the search params for the patterns collection.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		$query_params = parent::get_collection_params();
```

Comment:
If we don't support paginating, we should `unset` the `page` and `per_page` fields.

Reply:
🤔  I'm not sure if we will or not, but there's no harm in disabling now. We can always enable if we discover we do need it.

Reply:
removed in 447d228a4f